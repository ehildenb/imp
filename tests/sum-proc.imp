int finalSum;

def sum ( n ) {
    int s;
    s = 0;
    while (0 <= n) {
        s = s + n;
        n = n - 1;
    }
    return s;
}

finalSum = sum ( 10 );
