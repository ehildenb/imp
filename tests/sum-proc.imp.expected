<imp>
  <k>
    .
  </k>
  <mem>
    finalSum |-> 55
  </mem>
  <procs>
    sum |-> def sum ( n , .Ids ) { int s , .Ids ; s = 0 ; while ( ! n <= 0 ) { s = s + n ; n = n - 1 ; } return s ; }
  </procs>
  <callStack>
    .List
  </callStack>
</imp>
